source/GameWindow.cpp

```diff
@@ -0,0 +1,375 @@
+/* GameWindow.cpp
+Copyright (c) 2014 by Michael Zahniser
+
+Provides the ability to create window objects. Like the main game window.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Files.h"
+#include "GameWindow.h"
+#include "gl_header.h"
+#include "ImageBuffer.h"
+#include "Preferences.h"
+#include "Screen.h"
+
+#include <cstring>
+#include <SDL2/SDL.h>
+#include <string>
+#include <sstream>
+
+using namespace std;
+
+namespace {
+	SDL_Window *mainWindow;
+	SDL_GLContext context;
+	int width = 0;
+	int height = 0;
+	//int monitorHz;
+	bool hasSwizzle;
+		
+	// Logs SDL errors and returns 1 if found
+	bool checkSDLerror(const string &context)
+	{
+		string message = SDL_GetError();
+		if(!message.empty())
+		{
+			Files::LogError("(SDL message: \"" + message + "\")");
+			SDL_ClearError
```

Comment:
Did you mean to perform a function call here (and in several other places)?

Reply:
Wow. I really borked it up good with that last patch. Somehow I thought I had tested this but I must have run the previous build. I resolved all errors and warnings. Double checked the build.